### DISCUSSION TITLE
Different versions for implementations

### DISCUSSION THREAD

**Discussion by cyrusmsk** [jinyus/related_post_gen#484] [General]
Hi.

Currently most of the langs have only 1 version in each table: single core and multithreaded.
However, for D we have regular D solution (which is very readable and has clean and easy to understand code) and D_v2 solution (which is more performant, but applying different approach for calculation). This was done to show other community of the benchmark some ways that could be used to improve performance of the single core solution and the difference that you can get with more advanced techniques in code (D_v2) in comparison with naive and simple implementation (D).
However, Rust for example has only 1 solution, but latest commit was based on ideas from D_v2.

Now one of the member of D community has prepared another version, which is using another idea. It could be D_v3.. or we could just move everything to D solution.

Many different techniques were applied in the past and as I already was mentioned in one of the issues, that it will be interesting to save this history of improvements with separate techniques - but it is hard.

So the question is to @jinyus mostly - do you want us to split (if the algos are significantly different) versions in the table (then I would like to see Rust_v2) or just use only 1 (the fastest available) version to not increase the complexity in the resulted table?

**Comment by zupat** [comment]
The separation was only done for aesthetic reasons so I'm not against merging the 2, and probably include the regular version in the repo with the name `idiomatic.d`
